nistats/regression.py

```diff
@@ -347,41 +355,19 @@ def __init__(self, results):
         # put this as a parameter of LikelihoodModel
         self.df_resid = self.df_total - self.df_model
 
+        self.wdesign = results.model.wdesign
+
     def logL(self, Y):
         """
         The maximized log-likelihood
         """
-        raise ValueError('can not use this method for simple results')
+        raise ValueError('minimize_memory should be set to False to make residuals or predictions.')
 
-    def resid(self, Y):
+    def resid(self):
         """
         Residuals from the fit.
         """
-        return Y - self.predicted
-
-    def norm_resid(self, Y):
-        """
-        Residuals, normalized to have unit length.
-
-        Notes
-        -----
-        Is this supposed to return "stanardized residuals,"
-        residuals standardized
-        to have mean zero and approximately unit variance?
-
-        d_i = e_i / sqrt(MS_E)
-
-        Where MS_E = SSE / (n - k)
-
-        See: Montgomery and Peck 3.2.1 p. 68
-             Davidson and MacKinnon 15.2 p 662
-        """
-        return self.resid(Y) * positive_reciprocal(np.sqrt(self.dispersion))
+        raise ValueError('minimize_memory should be set to False to make residuals or predictions.')
```

Comment:
@bthirion @jdkent @Gilles86 
I'm still confused about the suitability of this, and other such methods. It feels like 'logL()', `resid()` and `norm_resid()` are in effect, traps; waiting for unsuspecting users to call them and watch their programs fail.

If these methods do not have anything to say, then we should simply remove all of them, or convert this into a message rather than an error.

Reply:
Agreed.

Reply:
I believe the motivation for this was to provide a hint to the user on how to get residuals from their model instead of an attribute error, but I see this check is [handled here](https://github.com/jdkent/nistats/blob/1a8e0c599673cdd687b94488a6db9d585708020f/nistats/first_level_model.py#L617). I'm fine with removing those methods.